### PR TITLE
EL-1764: Add ability to print FALA results page

### PR DESCRIPTION
**Pull request by MazOneTwoOne**
## What does this pull request do?

- update gulpfile
- add in css
- place button on results page
- rename CSS
- add in JS
- don't print footer, banner, beta banner, google translate
- add css so that content after <a> links don't print
- wrap pagination with `govuk-!-display-none-print` class

## Any other changes that would benefit highlighting?

Intentionally left blank.

## Checklist

- [ ] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title"

### CODE REVIEW

**Review by willc-work — DISMISSED** [review]

**Review by willc-work — APPROVED** [review]
I am happy with the css and the html but would appreciate @psweeting1 reviewing the JS. Approving t because I have no doubt it works as intended

File: fala/assets-src/sass/main.scss [hunk 1]
```diff
@@ -130,3 +152,37 @@ body.fala-dev {
   font-size: 19px;
   -webkit-font-smoothing: antialiased;
 }
+
+.print-link__background, .print-link__button {
```
Comment (psweeting1) [inline fala/assets-src/sass/main.scss:156]:
```suggestion
.print-link {
  &__background,
  &__button {
    background: none;
    background-size: 16px 16px;
    padding-top: 10px;
    padding-right: 10px;
    padding-bottom: 10px;
    text-decoration: none;
  }

  &__button {
    border: 1px solid govuk-colour("dark-grey");
    color: $govuk-link-colour;
    cursor: pointer;
    margin: govuk-spacing(0);

    &:hover {
      background-color: govuk-colour("light-grey");
    }

    &:focus {
      @include govuk-focused-text;
      background-color: $govuk-focus-colour;
      border-color: transparent;
    }
  }

  &__icon {
    width: 16px;
    height: 18px;
    vertical-align: middle;
    margin-right: 8px;
    margin-left: 5px;
  }
}
```

File: fala/templates/adviser/adviser_base.html [hunk 2]
```diff
@@ -97,4 +97,13 @@ <h1 class="govuk-heading-xl govuk-!-margin-top-7">
       );
     });
   </script>
+
+  <script nonce="{{request.csp_nonce}}">
+    const printResultsPage = document.getElementById("printButton");
```
Comment (psweeting1) [inline fala/templates/adviser/adviser_base.html:102]: Don't do this please, bad practice. Please create a `print.js` file and concat it with the govuk js file being imported into the project. 

Keeping JavaScript in separate files makes the code easier to read, maintain, and debug. It allows developers to work on the HTML structure and JavaScript logic independently, leading to cleaner and more organised code.

JavaScript code in a separate file can be reused across multiple pages of a website or even across different projects. This reduces code duplication and makes updates easier.

Finally debugging. I cannot debug inline JS.

File: fala/templates/adviser/results.html [hunk 3]
```diff
@@ -54,6 +54,18 @@ <h1 class="govuk-heading-xl">Search results</h1>
         }) }}
       </form>
     </div>
+
+    {% if data.count >= 1 %}
+      <div class="printing-button">
+        <div class="print-link__background govuk-!-display-none-print govuk-!-margin-top-3 govuk-!-margin-bottom-5">
+          <button id="printButton" class="govuk-link govuk-body-s print-link__button">
+            <img src="/static/images/icon-print.png" alt="Print Icon" class="print-link__icon">
```
Comment (psweeting1) [inline fala/templates/adviser/results.html:62]: Why could you not have the icon as a background import? It's actually better if it is a background image.

See how HMRC and GOVUK do print in their service. They use CSS to apply the icon to the background. Makes it more browser friendly and better for performance (Though that is tiny). 

I imagine as well for accessibility a screen reader will also read out there is an image when in fact icons shouldn't really be read out by screen readers.

File: fala/assets-src/sass/main.scss [hunk 4]
```diff
@@ -40,14 +46,30 @@ $laa-dev-colour: govuk-colour("pink");
   }
 }
 
-.no-js .laa-cookie-policy {
+.laa-cookie-policy {
```
Comment (psweeting1) [inline fala/assets-src/sass/main.scss:49]:
```suggestion
.laa-cookie-policy {
  display: none;

  .js-enabled & {
    display: block;
  }
}

.fala-cookie-banner {
  display: none;

  .js-enabled & {
    display: block;
  }
}

.printing-button {
  display: none;

  .js-enabled & {
    display: block;
  }
}
```

File: fala/templates/adviser/adviser_base.html (hunk 2 above)
Comment (MazOneTwoOne) [reply fala/templates/adviser/adviser_base.html:102]: Will do. It's such a small bit of JS that I thought I could get away with it, but if i's bad practice will put into it's own file. 👍

File: fala/templates/adviser/results.html (hunk 3 above)
Comment (MazOneTwoOne) [reply fala/templates/adviser/results.html]: Have adjusted.

**Review by psweeting1 — APPROVED** [review]
LGTM

**Pull request merged**